feat: added skipping of package.json check if package is whitelisted (resolves #118)

mat-sz opened the pull request:
I'm setting default values for name and version, just to not cause issues further down the road.

Resolves #118.

JustinBeckwith commented on src/checker.ts:
```diff
@@ -315,7 +315,14 @@ export class LicenseChecker extends EventEmitter {
+            ...json,
+          }
+        : ensurePackageJson(json);
     if (!packageName) {
```
To avoid the extra check above, could we pull this up to the start of the function and instead say:
```js
packageName = packageName || pj.name;
```

mat-sz replied:
pj is not defined at the start of the function. I may be missing the point there though.

JustinBeckwith commented on src/checker.ts:
```diff
@@ -315,7 +315,14 @@ export class LicenseChecker extends EventEmitter {
     ...parents: string[]
   ): Promise<void> {
-    const pj: PackageJson = ensurePackageJson(json);
+    const pj: PackageJson =
```
Instead of the ternary, could we instead do something like:

```js
packageName = packageName || json.name;
const isWhitelisted = this.isPackageWhitelisted(packageName);
if (isWhitelisted) {
  json.version = semver.valid(json.version) ? json.version : '0.0.0'
  // any other corrections we may want to make in the future
}
```

mat-sz commented:
@JustinBeckwith Thank you for the suggestions, I've resolved the issues and added a test.

Also, I've moved the function away from reliance on JS `undefined` casting into a string `"undefined"` by adding an explicit `|| "undefined"`.

mat-sz commented:
@JustinBeckwith Apologies for the repeated mention, is there anything else required for this PR?

JustinBeckwith commented:
Thanks for sticking with me on the changes!
